src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,57 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static int GROWTH_FACTOR;
+    private T[] stack;
+    private int size;
+    private int capacity;
 
+    public MateStack() {
+        GROWTH_FACTOR = 2;
+        capacity = 10;
+        stack = (T[]) new Object[capacity];
+        size = 0;
+    }
+
+    public MateStack(int defaultCapacity, int growthFactor) {
+        GROWTH_FACTOR = growthFactor;
```

Comment:
As `GROWTH_FACTOR` is static, setting it here will affect all instances of `MateStack`. Consider making `GROWTH_FACTOR` a non-static instance variable to avoid this issue.